fluent-syntax/src/stream.js

```diff
@@ -24,6 +26,7 @@ export class ParserStream {
     }
 
     this.index++;
+    this.checkLineBreak();
```

Comment:
Thanks, that's very helpful. I agree about the separations of concerns. Also, it seems like it would be easier to provide a helper for now and add `lineno` to the AST later than the other way around.